<?xml version="1.0" encoding="utf-8"?>
<launch>
  <!-- Include the smb_gazebo.launch file -->
  <include file="$(find smb_gazebo)/launch/smb_gazebo.launch">
    <!-- Pass arguments to smb_gazebo.launch -->
    <arg name="headless" default="true"/>
    <arg name="model_path" default="$(find smb_gazebo)/"/>
    <arg name="robot_namespace" default=""/>
    <arg name="robot_model_name" default="smb"/>
    <arg name="enable_ekf" default="true"/>
    <arg name="world" default="robocup14_spl_field"/>
    <arg name="world_file" default="$(find eth_exec_3)/worlds/singlePillar.world"/>
    <arg name="laser_enabled" default="true"/>
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.4"/>
    <arg name="roll" default="0.0"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="0.0"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="true"/>
    <arg name="laser_scan_min_height" default="-0.2"/>
    <arg name="laser_scan_max_height" default="1.0"/>
    <arg name="run_gui" default="true"/>
    <arg name="robot_description" default="robot_description"/>
  </include>

  <!-- Start the SMB Highlevel Controller Node specific to pillar detection -->
  <node name="pillar_detection" pkg="eth_exec_3" type="pillar_detection_node" output="screen">
    <param name="controller_gain_linear" value="-0.2"/>
    <param name="controller_gain_angular" value="1.0"/>
    <param name="target_distance" value="1.0"/>
  </node>
</launch>
